packages/rocketchat-livechat/imports/server/rest/messages.js

```diff
@@ -0,0 +1,64 @@
+import LivechatVisitors from '../../../server/models/LivechatVisitors';
+
+RocketChat.API.v1.addRoute('livechat/messages', {
+	post() {
+		if (!RocketChat.authz.hasPermission(this.userId, 'view-livechat-manager')) {
+			return RocketChat.API.v1.unauthorized();
+		}
+
+		if (!this.bodyParams.visitor) {
+			return RocketChat.API.v1.failure('Body param "visitor" is required');
+		}
+		if (!this.bodyParams.visitor.token) {
```

Comment:
`this.bodyParams.visitor.token` doesn't work. You should use camelCased in the field names, such as `this.bodyParams.visitorToken`.

Reply:
@renatobecker the payload used in message post was based in the payload send to webhook.

This is the expected POST for this endpoint:
```
curl -v -H "Content-type: application/json" -H "${H_USERID}" -H "${H_AUTHTOKEN}" "http://localhost:3000/api/v1/livechat/messages" -d'{
  "visitor": {
    "token":"04aa15e2-31ed-11e8-ac97-93976ab3c67b",
    "name": "Guest Name",
    "email": "guest@mail.com",
    "phone": {
        "number": "+5511999999999"
    }
  },
  "messages": [
    {
       "msg": "Hi!!"
    }
  ]
}'
```

Reply:
Have you tried your code? we are talking about the name of a query parameter and the format you are using is not valid.

Reply:
Yes, we tried it. We are using a body parameter (JSON object) not a query parameter. And this format is to access field `token` inside `visitor` inside request body (application/json). This is why "visitor.token"

The "curl" below is working and will create a new guest as shown in the screenshot.
```
curl -s -H "Content-type: application/json" -H "${H_USERID}" -H "${H_AUTHTOKEN}" "http://localhost:3000/api/v1/livechat/messages" -d'{
  "visitor": {
    "token":"04aa15e2-31ed-11e8-ac97-93976ab3c67b",
    "name": "Guest Name",
    "email": "guest@mail.com",
    "phone": {
        "number": "+5511999999999"
    }
  },
  "messages": [
    {
       "msg": "Hi!!"
    }
  ]
}'
```

![screenshot-new-guest](https://user-images.githubusercontent.com/3268876/37989808-ceeede86-31da-11e8-834d-9049dd8f4c5e.png)

Reply:
If you want to try, here is the whole script:
https://gist.github.com/hmagarotto/6251f86e1f00035cc249c8ef05238a90

Reply:
OK, sorry. I was testing this in another way.